eo-runtime/src/main/eo/org/eolang/seq.eo

```diff
@@ -28,4 +28,20 @@
 
 # The object, when being dataized, dataizes all provided
 # "steps" (except the last one) and returns the last one.
-[steps] > seq /?
+[steps] > seq
+  if. > @
+    steps.length.eq 0
+    true
+    loop 0
+
+  [index] > loop
+    if. > @
+      and.
+        index.lt (steps.length.minus 1)
```

Comment:
@Chamber6821 let's move `steps.length.minus 1` out of `loop` and cache it
So it should be something like:
```
steps.lenght.minus 1 > max-len!
[index] > loop
  if. > @
    and.
      index.lt max-len
      ...
```

Reply:
This code leads to infinity loop (100% CPU load for many minutes)
```
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0

  steps.length.minus 1 > [] > max-len!
  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```
But code with out `!` works correctly. It is a bug :/

Reply:
@Chamber6821 works on my laptop:
```
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0
  steps.length.minus 1 > max-len!

  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```

Reply:
Your code works correctly with `eoc`, but leads to error in project `eo-runtime`

```
[ERROR] src/main/eo/org/eolang/seq.eo, line 37: [37:25] no viable alternative at input '# The object, when being dataized, dataizes all p
rovided\n# "steps" (except the last one) and returns the last one.\n[steps] > seq\n  INTif. > @\n    INTsteps.length.eq 0\n    true\n    
loop 0\nEXPONENT\n  steps.length.minus 1 > max-len': "  steps.length.minus 1 > max-len!"
[WARNING] src/main/eo/org/eolang/tuple.eo, line 48: Comment must be at least 64 characters long
[WARNING] src/main/eo/org/eolang/tuple.eo, line 71: Comment must be at least 64 characters long
[ERROR] net.sf.saxon.style.XSLMessage@a1325a2: Critical error identified:
  /home/Chamber6821/Projects/eo/eo-runtime/src/main/eo/org/eolang/seq.eo, 37: [37:25] no viable alternative at input '# The object, when 
being dataized, dataizes all provided\n# "steps" (except the last one) and returns the last one.\n[steps] &gt; seq\n  INTif. &gt; @\n    
INTsteps.length.eq 0\n    true\n    loop 0\nEXPONENT\n  steps.length.minus 1 &gt; max-len': "  steps.length.minus 1 &gt; max-len!";
[ERROR] #fatalError(): Processing terminated by xsl:message at line 31 in fail-on-critical.xsl; SystemID: file:///org/eolang/parser/fail-
on-critical.xsl; Line#: 31; Column#: 34
[ERROR] Processing terminated by xsl:message at line 31 in fail-on-critical.xsl
```

Reply:
If I replace `steps.length.minus 1 > max-len!` with `steps.length.minus 1 > [] > max-len!` that leads to infinity loop

Reply:
@Chamber6821 there's no need for `steps.length.minus 1 > [] > max-len!`. It's the same as:
```
[] > max-len
  steps.length.minus 1 > @
```

Reply:
@Chamber6821 this should work, with comment in front of `seq`:
```
# Seq comment here.
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0
  steps.length.minus 1 > max-len!

  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```

Reply:
The parser complains not about the comment (it is present in the code), but about the line `  steps.length.minus 1 > max-len!`

Reply:
@Chamber6821 works for me:
<img width="563" alt="image" src="https://github.com/objectionary/eo/assets/37025995/b843d6fa-4ddb-414f-be1a-cc0797b4331d">

Did you update you `master` branch?

Reply:
@Chamber6821 maybe you have some extra spaces on empty line between `max-len` and `loop`?

Reply:
Yo. Code 1 was be parsed successfully, but code 2 and code 3 was be parsed with error

Code 1:
```
# The object, when being dataized, dataizes all provided
# "steps" (except the last one) and returns the last one.
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0
  steps.length.minus 1 > max-len!

  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```

Code 2:
```
# The object, when being dataized, dataizes all provided
# "steps" (except the last one) and returns the last one.
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0

  steps.length.minus 1 > max-len!

  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```

Code 3:
```
# The object, when being dataized, dataizes all provided
# "steps" (except the last one) and returns the last one.
[steps] > seq
  if. > @
    steps.length.eq 0
    true
    loop 0

  # Max length.
  steps.length.minus 1 > max-len!

  [index] > loop
    if. > @
      and.
        index.lt ^.max-len
        or.
          (dataized (steps.at index)).as-bool
          true
      ^.loop
        index.plus 1
      steps.at index
```

Is it bug? May be platform dependent? (very strange for Java)

Reply:
@Chamber6821 there's no bug, it's our grammar rules. Is short - empty line is allowed only in front of named abstract object. Check [this](https://github.com/objectionary/eo/issues/1469)